langchain_tools_demo/main.py

```diff
@@ -118,34 +114,38 @@ async def chat_handler(request: Request, prompt: str = Body(embed=True)):
         )
 
     # Add user message to chat history
-    request.session["history"].append(message_to_dict(HumanMessage(content=prompt)))
-    user_agent = await get_agent(request.session, user_id_token=None)
+    request.session["history"].append({"type": "human", "data": {"content": prompt}})
+    ai = await get_agent(request.session, user_id_token=None, orchestration=None)
     try:
         print(prompt)
         # Send prompt to LLM
-        response = await user_agent.agent.ainvoke({"input": prompt})
+        response = await ai.invoke(prompt)
         # Return assistant response
         request.session["history"].append(
-            message_to_dict(AIMessage(content=response["output"]))
+            {"type": "ai", "data": {"content": response["output"]}}
         )
         return markdown(response["output"])
     except Exception as err:
         raise HTTPException(status_code=500, detail=f"Error invoking agent: {err}")
 
 
-async def get_agent(session: dict[str, Any], user_id_token: Optional[str]):
-    global user_agents
+async def get_agent(
+    session: dict[str, Any], user_id_token: Optional[str], orchestration: Optional[str]
```

Comment:
Why let orchestration be optional?

Reply:
We call for the get_agent function in 3 different endpoints. and within the 3 function calls, technically only one of them will be creating the agent. Hence, similar to user_id_token, I'm thinking of passing the orchestration variable when needed.